lib/srv/regular/sftp.go

```diff
@@ -189,22 +189,51 @@ func (s *sftpSubsys) Start(ctx context.Context,
 				return
 			}
 
-			var sftpEvent apievents.SFTP
-			err = jsonpb.UnmarshalString(eventStr[:len(eventStr)-1], &sftpEvent)
+			// If the next byte is a NULL byte, this event is a SFTPSummary event
+			var lastEvent bool
+			var event apievents.AuditEvent
+			nextByte, err := r.Peek(1)
 			if err != nil {
-				s.log.WithError(err).Warn("Failed to unmarshal SFTP event.")
-				continue
-			}
+				s.log.WithError(err).Warn("Failed to peek SFTP event file.")
+			} else if nextByte[0] == 0x0 {
```

Comment:
This feels kinda hacky, do we need to invent this custom unmarshal scheme with a magic null byte? Can't we just unmarshal the event header first and look at the event type like we do with other resources?

Reply:
I'll try that tomorrow, that is probably a better idea in hindsight